.github/workflows/build.yml

```diff
@@ -45,10 +46,38 @@ jobs:
           name: ${{ matrix.asset_name }}
           path: dist/${{ matrix.artifact_name }}
 
+  test_run:
```

Comment:
I do not really like to test this in the build workflow. I mean, we release and if any OS fails we don't upload any binary. It would be better to test this before releasing. What about adding a test build workflow which is run in master (which should be green before releasing) and removing it from here?

Reply:
hm, how can we do this on a per-case basis? maybe via a specific activity type https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release?

Reply:
What do you mean? running it only on a release PR?

My proposal was to create a `build_test.yml` which is run on master:

```
on:
  push:
    branches:
      - master

Reply:
Instead of duplicating steps in a new workflow file my new proposal uses `if` checks to `test_run` on push to master or to `zip` on release.